pkg/converters/utils/services.go

```diff
@@ -73,27 +73,92 @@ type Endpoint struct {
 }
 
 // CreateEndpoints ...
-func CreateEndpoints(cache types.Cache, svc *api.Service, svcPort *api.ServicePort) (ready, notReady []*Endpoint, err error) {
+func CreateEndpoints(cache types.Cache, svc *api.Service, svcPort *api.ServicePort, useEndpointSlices bool) (ready, notReady []*Endpoint, err error) {
```

Comment:
This func seems more like a builder/factory now, consider the snippet below as a way to split specific behavior to specific new funcs.

```go
	switch {
	case svc.Spec.Type == api.ServiceTypeExternalName:
		ready, err := createEndpointsExternalName(cache, svc, svcPort)
		return ready, nil, err
	case useEndpointSlices:
		endpoints, err := cache.GetEndpointSlices(svc)
		if err != nil {
			return nil, nil, err
		}
		return createEndpointSlices(endpoints, svcPort)
	default:
		endpoints, err := cache.GetEndpoints(svc)
		if err != nil {
			return nil, nil, err
		}
		return createEndpoints(endpoints, svcPort)
	}
```

```go
func createEndpoints(endpoints *api.Endpoints, svcPort *api.ServicePort) (ready, notReady []*Endpoint, err error)
```

```go
func createEndpointSlices(endpointSlices []*discoveryv1.EndpointSlice, svcPort *api.ServicePort) (ready, notReady []*Endpoint, err error)
```